### PR TITLE
Modify Authorized API UIs to support new API Authorization support

### PR DESCRIPTION
**Pull request by SujanSanjula96**
Related Issue - https://github.com/wso2/product-is/issues/16344

### CODE REVIEW

File: apps/console/src/features/applications/api/use-scopes-of-api-resources.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { AuthorizedPermissionListItemInterface } from "../models/api-authorization";
+
+/**
+ * Hook to get the scopes of an API resource.
+ *
+ * @param appId - application ID.
+ * @returns SWR response object containing the data, error, isLoading, mutate.
+ * @throws `IdentityAppsApiException`
+ */
+const useScopesOfAPIResources = <Data = AuthorizedPermissionListItemInterface[], Error = RequestErrorInterface>(
```
Comment (Achintha444) [inline apps/console/src/features/applications/api/use-scopes-of-api-resources.ts:35]: shall we move this to the `api-authoirzation.ts` file

File: apps/console/src/features/applications/api/use-subscribed-api-resources.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { HttpMethods } from "@wso2is/core/models";
```
Comment (Achintha444) [inline apps/console/src/features/applications/api/use-subscribed-api-resources.ts:19]: shall we move this to the api-authoirzation.ts file

File: apps/console/src/features/applications/api/use-scopes-of-api-resources.ts (hunk 1 above)
Comment (SujanSanjula96) [reply apps/console/src/features/applications/api/use-scopes-of-api-resources.ts:35]: Moved the GET hooks to dedicated modules with the suggestion given in a previous PR (https://github.com/wso2/identity-apps/pull/4271/files#r1367843706).

File: apps/console/src/features/applications/api/use-subscribed-api-resources.ts (hunk 2 above)
Comment (SujanSanjula96) [reply apps/console/src/features/applications/api/use-subscribed-api-resources.ts:19]: Moved the GET hooks to dedicated modules with the suggestion given in a previous PR (https://github.com/wso2/identity-apps/pull/4271/files#r1367843706).

File: apps/console/src/features/applications/components/api-authorization/scope-form.tsx [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+
+    /**
+     * Handles the disabled state of the select all button.
+     */
+    useEffect(() => {
+        setIsSelectAllDisabled(currentAPIResourceScopeListData?.length === selectedScopes?.length);
+    }, [ selectedScopes, subscribedAPIResource ]);
+
+    /**
+     *
```
Comment (Achintha444) [inline apps/console/src/features/applications/components/api-authorization/scope-form.tsx:180]: Add a coment.

File: apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+            } 
+        ];
+    };
+
+    /**
+     * Resolves the content of the subscribed API Resources list item.
+     */
+    const resolveSubscribedAPIResourcesListContent = (subscribedAPIResource: AuthorizedAPIListItemInterface):
+        ReactElement => (
+
```
Comment (Achintha444) [inline apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx:335]:
```suggestion
```

File: apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+        <>
```
Comment (Achintha444) [inline apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx:336]: Remove this component as there is only one child

File: apps/console/src/features/applications/components/api-authorization/scope-form.tsx (hunk 3 above)
Comment (SujanSanjula96) [reply apps/console/src/features/applications/components/api-authorization/scope-form.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4300/commits/dc52d8ae5a27ca520a4ea0bf62bba8b5899ceb74

File: apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx (hunk 4 above)
Comment (SujanSanjula96) [reply apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4300/commits/dc52d8ae5a27ca520a4ea0bf62bba8b5899ceb74

File: apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx (hunk 5 above)
Comment (SujanSanjula96) [reply apps/console/src/features/applications/components/api-authorization/subscribed-api-resources.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4300/commits/dc52d8ae5a27ca520a4ea0bf62bba8b5899ceb74

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**